tests/rptest/tests/retention_policy_test.py

```diff
@@ -262,9 +262,11 @@ def test_shadow_indexing_default_local_retention(self,
                             self.segment_size)
 
         if expect_deletion:
+            # Wait up to just a bit longer than the 10s default config value
+            # for `cloud_storage_upload_loop_max_backoff_ms`.
             wait_until(
                 lambda: self.segments_removed(self.default_retention_segments),
-                timeout_sec=5,
+                timeout_sec=15,
```

Comment:
i wonder how frequent this pattern occurs? it might be cool to have something like
```
wait_until_config(predicate, timeout_from_config="cloud_storage_upload_loop_max_backoff_ms")
```

some day

Reply:
Yeah I think the "right" approach would be to manually set `cloud_storage_upload_loop_max_backoff_ms` and wait for some multiple of that constant. It reduces the authenticity a bit of the test, but it at least avoids these kinds of test bugs and might reduce runtimes.